src/WebJobs.Script/ExtensionBundle/ExtensionBundleHttpExtensions.cs

```diff
@@ -0,0 +1,73 @@
+// Centralized helper for reading and sanitizing the x-azure-ref header for Extension Bundle downloads.
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Script.ExtensionBundle
+{
+    internal static class ExtensionBundleHttpExtensions
+    {
+        internal const string AzureRefHeaderName = "x-azure-ref";
+        private const int MaxAzureRefLength = 128;
+
+        internal static string GetAzureRef(this HttpResponseMessage response)
+        {
+            try
+            {
+                if (response == null)
+                {
+                    return null;
+                }
+
+                if (response.Headers != null &&
+                    response.Headers.TryGetValues(AzureRefHeaderName, out var values))
+                {
+                    return Sanitize(values.FirstOrDefault());
+                }
+
+                if (response.Content?.Headers != null &&
+                    response.Content.Headers.TryGetValues(AzureRefHeaderName, out var contentValues))
+                {
+                    return Sanitize(contentValues.FirstOrDefault());
+                }
+            }
+            catch
+            {
+                // Ignore header parsing issues.
+            }
+
+            return null;
+        }
+
+        private static string Sanitize(string value)
```

Comment:
Can you share more information about the need to trim and sanitize this string? Don't we want to preserve reference? Did you run into issues with what they return?

Reply:
Mainly defensive logging. However, it is fixed format and ok to remove sanitization. I'll remove it.